budget_control_advance_clearing/models/hr_expense.py

```diff
@@ -13,13 +13,17 @@ class HRExpenseSheet(models.Model):
     )
 
     def write(self, vals):
-        """ Clearing for its Advance """
+        """ Clearing for its Advance and Cancel payment expense """
+        cancel_state = self.state == "cancel" and True or False
```

Comment:
Try to use this way, it is more explicit (and many time save some bugs)
```suggestion
        cancel_state = True if self.state == "cancel" else False
```

Reply:
self is `multi`, this `self.state` can cause singleton error?

Reply:
พี่ว่าควรแยก filtered docs ที่เป็น cancel state ออกจากส่วนอื่น แล้วทำทีเดียว

Reply:
พี่ว่าควรแยก filtered docs ที่เป็น cancel state ออกจากส่วนอื่น แล้วทำทีเดียว

Reply:
@Saran440 you seem always forgot about singleton error, please be very careful. I..e,
https://github.com/OCA/hr-expense/pull/56/commits/77e195ef4d7039d126f5a74065dee78588555291

Reply:
@Saran440 you seem always forgot about singleton error, please be very careful. I..e,
https://github.com/OCA/hr-expense/pull/56/commits/77e195ef4d7039d126f5a74065dee78588555291